src/encoder/encodersndfileflac.cpp

```diff
@@ -30,10 +59,38 @@ void EncoderSndfileFlac::setEncoderSettings(const EncoderSettings& settings)
     m_compression = static_cast<double>(settings.getCompression()) / 8.0;
 }
 
+void EncoderSndfileFlac::encodeBuffer(const CSAMPLE* pBuffer, const int iBufferSize) {
+    if (m_pClampBuffer) {
+        SINT numSamplesLeft = iBufferSize;
+        while (numSamplesLeft > kEncBufferSize) {
+            convertFloat32ToIntFormat(m_pClampBuffer.get(),
+                    pBuffer,
+                    kEncBufferSize,
+                    m_sfInfo.format);
+            sf_write_int(m_pSndfile, m_pClampBuffer.get(), kEncBufferSize);
+            pBuffer += kEncBufferSize;
+            numSamplesLeft -= kEncBufferSize;
+        }
+        convertFloat32ToIntFormat(m_pClampBuffer.get(), pBuffer, numSamplesLeft, m_sfInfo.format);
```

Comment:
These instruction should only appear inside the loop instead of duplicating the code.

`numSamplesToWrite = math_min(numSamplesLeft, kEncBufferSize);`